nodejs/server.js

```diff
@@ -167,4 +169,116 @@ app.all('/', function (req, res) {
 
 });
 
-app.listen(argv.port);
+let server = require('http').createServer({
+    perMessageDeflate: {
+        zlibDeflateOptions: {
+            // See zlib defaults.
+            chunkSize: 1024,
+            memLevel: 7,
+            level: 3
+        },
+        zlibInflateOptions: {
+            chunkSize: 10 * 1024
+        },
+        // Other options settable:
+        clientNoContextTakeover: true, // Defaults to negotiated value.
+        serverNoContextTakeover: true, // Defaults to negotiated value.
+        serverMaxWindowBits: 10, // Defaults to negotiated value.
+        // Below options specified as default values.
+        concurrencyLimit: 10, // Limits zlib concurrency for perf.
+        threshold: 1024 // Size (in bytes) below which messages
+        // should not be compressed.
+    }
+});
+
+// Also mount the app here
+server.on('request', app);
+
+const websocketEvent = {
+    url: 'http://localhost:8000/websocket'
+  } 
+
+const inactiveEvent = {
+    url: 'http://localhost:8000/inactive'
+  } 
+
+
+// Create web socket server on top of a regular http server
+let wss = new WSServer({
+    server: server
+});
+
+
+function noop() {}
+
+function heartbeat() {
+  this.isAlive = true;
+}
+
+
+wss.on('connection', function connection(ws) {
+    // On successful request, there's no body returned 
+    request(websocketEvent, (err, res)=>{
+        if (err || res.statusCode!=200){
+            if (err){
+            console.log(err)
+            } else{
+            console.log("Unexpected response")
+            }
+            ws.send("Error")
+            return 
+        }
+    })
+    ws.isAlive = true;
+    ws.on('pong',heartbeat);
+
+    const interval = setInterval(function ping() {
```

Comment:
Do you want to register this interval checker for each connection created, or only once since it checks all connections itself? I think this belongs outside the connection handler

Reply:
You're right, thanks for pointing that. I'll move it out